File: backend/app/schemas/settlement.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class SettlementCreate(BaseModel):
    group_id: UUID

    payer_id: UUID

    receiver_id: UUID

    amount: Decimal = Field(gt=0)

    note: str | None = Field(
        default=None,
        max_length=255,
    )


class SettlementResponse(BaseModel):
    id: UUID

    group_id: UUID

    payer_id: UUID
    payer_name: str

    receiver_id: UUID
    receiver_name: str

    amount: Decimal

    note: str | None

    settled_at: datetime

File: frontend/lib/features/settlements/models/settlement.dart
class Settlement {
  final String id;

  final String payerId;
  final String payerName;

  final String receiverId;
  final String receiverName;

  final double amount;
  final String? note;

  final DateTime settledAt;

  Settlement({
    required this.id,
    required this.payerId,
    required this.payerName,
    required this.receiverId,
    required this.receiverName,
    required this.amount,
    required this.note,
    required this.settledAt,
  });

  factory Settlement.fromJson(
    Map<String, dynamic> json,
  ) {
    return Settlement(
      id: json["id"],
      payerId: json["payer_id"],
      payerName: json["payer_name"],
      receiverId: json["receiver_id"],
      receiverName: json["receiver_name"],
      amount: double.parse(
        json["amount"].toString(),
      ),
      note: json["note"],
      settledAt: DateTime.parse(
        json["settled_at"],
      ),
    );
  }
}